### PR TITLE
chore: added jsdoc/check-tag-names to ESLint config

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
**Context**: Part of part of #648, porting old [TSLint](https://palantir.github.io/tslint) rules to [ESLint](https://eslint.org/) equivalents. This finishes migrating `no-redundant-jsdoc2` to `jsdoc/check-tag-names`.

This finishes the work in #65080, #65315, and #65445 by finally checking the rule into the ESLint config. Which I also converted to `.cjs` so I could add comments about trimming down that big list of allowed tags.

Also adds a stub `tsconfig.json` so [`project: true`](https://typescript-eslint.io/architecture/parser#project) always has a fallback.

### CODE REVIEW

**Comment by sandersn** [comment]
Here are the 21 packages with failures for a local run on all packages:

      1:Error in ember
     12:Error in ember/v3
     23:Error in ember__debug
     34:Error in ember__debug/v3
     45:Error in gulp-help-doc
     58:Error in ink-table
     69:Error in ink-testing-library
     80:Error in ink-text-input
     91:Error in node/v16
    110:Error in node
    130:Error in node/v18
    150:Error in openfin/v50
    296:Error in openfin
    434:Error in openui5
    449:Error in react-blessed
    460:Error in react-native/v0.71
    477:Error in react-reconciler
    488:Error in subset-font
    500:Error in vhtml
    511:Error in xmpp__xml
    522:Error in yeoman-test

The ember packages all have a tag `@ember/debug`; looks like a mis-parse which might be hard to avoid.

File: .eslintrc.cjs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    plugins: ['@typescript-eslint', 'jsdoc'],
+    rules: {
+        'dt-header': 'error',
+        'export-just-namespace': 'error',
+        'jsdoc/check-tag-names': [
+            'error',
+            {
+                // TODO: Some (but not all) of these tags should likely be removed from this list.
+                // Additionally, some may need to be contributed to eslint-plugin-jsdoc.
+                definedTags: [
```
[the remark this answers is not in the document]
Comment (JoshuaKGoldberg) [reply .eslintrc.cjs:17]: It's the tags that are used on DT and explicitly allowed. https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/check-tag-names.md#definedtags

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-reconciler ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-reconciler@latest/))
was missing the following properties:
1. attemptContinuousHydration
1. attemptDiscreteHydration
1. attemptHydrationAtCurrentPriority
1. attemptSynchronousHydration
1. batchedUpdates

<details>
<summary>as well as these 30 other properties...</summary>
<p>createComponentSelector, createContainer, createHasPseudoClassSelector, createHydrationContainer, createPortal, createRoleSelector, createTestNameSelector, createTextSelector, deferredUpdates, discreteUpdates, findAllNodes, findBoundingRects, findHostInstance, findHostInstanceWithNoPortals, findHostInstanceWithWarning, flushControlled, flushPassiveEffects, flushSync, focusWithin, getCurrentUpdatePriority, getFindAllNodesFailureDescription, getPublicRootInstance, injectIntoDevTools, isAlreadyRendering, observeVisibleRects, registerMutableSourceForHydration, runWithPriority, shouldError, shouldSuspend, updateContainer</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 30799962864493f80931672691c03eedac8d9d3c
</p>

File: types/ember/test/debug.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ const {
 const assert: typeof Ember.assert = Ember.assert;
 
 /**
- * @ember/debug tests
+ * `@ember/debug` tests
```
Comment (JoshuaKGoldberg) [inline types/ember/test/debug.ts:14]: Pending https://github.com/gajus/eslint-plugin-jsdoc/issues/1116, this seemed like the most straightforward resolution.

**Comment by JoshuaKGoldberg** [comment]
Note that the test failures in [eb565bf](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65660/commits/eb565bf2133b2b8e2bd413906bd2c7d92f9a3783) are unrelated to this PR. The later commit [3079996](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65660/commits/30799962864493f80931672691c03eedac8d9d3c) undid changes to those packages. I'd only touched the packages to make sure that they don't have ESLint complaints.

```plaintext
Error in react-blessed
Error: Errors in typescript@4.4 for external dependencies:
Error: ../react/jsx-runtime.d.ts(4,34): error TS2694: Namespace 'React.JSX' has no exported member 'ElementType'.

    at testTypesVersion (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:235:11)
    at runTests (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:197:7)


Error in yeoman-test
Error: 
A module look-up failed, this often occurs when you need to run `npm install` on a dependent module before you can lint.

Before you debug, first try running:

   npm install --prefix /home/runner/work/DefinitelyTyped/DefinitelyTyped/types/inquirer

Then re-run. Full error logs are below.

Errors in typescript@5.2 for external dependencies:
Error: ../inquirer/index.d.ts(16,28): error TS2307: Cannot find module 'rxjs' or its corresponding type declarations.
Error: ../inquirer/lib/prompts/base.d.ts(2,28): error TS2307: Cannot find module 'rxjs' or its corresponding type declarations.
Error: ../inquirer/lib/prompts/editor.d.ts(2,39): error TS2307: Cannot find module 'rxjs' or its corresponding type declarations.
Error: ../inquirer/lib/utils/events.d.ts(2,28): error TS2307: Cannot find module 'rxjs' or its corresponding type declarations.

    at testTypesVersion (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:235:11)
    at runTests (/home/runner/work/DefinitelyTyped/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:197:7)


Error in artillery
Error: Errors in typescript@5.2 for external dependencies:
Error: node_modules/cacheable-request/dist/index.d.ts(3,8): error TS1259: Module '"node:events"' can only be default-imported using the 'esModuleInterop' flag
Error: node_modules/got/dist/source/core/index.d.ts(484,39): error TS2694: Namespace '"/home/runner/work/DefinitelyTyped/DefinitelyTyped/types/artillery/node_modules/cacheable-request/dist/index"' has no exported member 'StorageAdapter'.
Error: node_modules/got/dist/source/core/index.d.ts(747,39): error TS2694: Namespace '"/home/runner/work/DefinitelyTyped/DefinitelyTyped/types/artillery/node_modules/cacheable-request/dist/index"' has no exported member 'StorageAdapter'.
Error: node_modules/got/dist/source/core/index.d.ts(779,39): error TS2694: Namespace '"/home/runner/work/DefinitelyTyped/DefinitelyTyped/types/artillery/node_modules/cacheable-request/dist/index"' has no exported member 'StorageAdapter'.
```

**Review by sheetalkamat — APPROVED** [review]

**Comment by JoshuaKGoldberg** [comment]
Ready to merge

**Pull request merged**